libraries/testing/include/eosio/testing/tester.hpp

```diff
@@ -392,46 +461,78 @@ namespace eosio { namespace testing {
          vcfg.reversible_guard_size = 0;
          vcfg.contracts_console = false;
 
-         vcfg.genesis.emplace();
-         vcfg.genesis->initial_timestamp = fc::time_point::from_iso_string("2020-01-01T00:00:00.000");
-         vcfg.genesis->initial_key = get_public_key( config::system_account_name, "active" );
+         genesis_state genesis;
+         genesis.initial_timestamp = fc::time_point::from_iso_string("2020-01-01T00:00:00.000");
+         genesis.initial_key = get_public_key( config::system_account_name, "active" );
 
          for(int i = 0; i < boost::unit_test::framework::master_test_suite().argc; ++i) {
             if(boost::unit_test::framework::master_test_suite().argv[i] == std::string("--wavm"))
                vcfg.wasm_runtime = chain::wasm_interface::vm_type::wavm;
             else if(boost::unit_test::framework::master_test_suite().argv[i] == std::string("--wabt"))
                vcfg.wasm_runtime = chain::wasm_interface::vm_type::wabt;
          }
-         return vcfg;
+         return {vcfg, genesis};
       }
 
       validating_tester(const flat_set<account_name>& trusted_producers = flat_set<account_name>()) {
-         vcfg = default_config(tempdir);
+         auto def_conf = default_config(tempdir);
 
+         vcfg = def_conf.first;
          vcfg.trusted_producers = trusted_producers;
 
-         FC_ASSERT(vcfg.genesis, "Should have a valid genesis_state");
-         validating_node = std::make_unique<controller>(vcfg, fc::optional<chain_id_type>(vcfg.genesis->compute_chain_id()), make_protocol_feature_set());
-         validating_node->add_indices();
-         validating_node->startup( []() { return false; } );
+         validating_node = create_validating_node(vcfg, def_conf.second, true);
 
          init();
```

Comment:
It looks like it should be init (def_conf.second), but not following your words here.  I'm going to make it that and let me know if we need to discuss more.

Reply:
(Should have looked more before I wrote), so I want to handle this using the previously unneeded genesis variable and creating the validating node after the fact.  Those this will handle it in a different order than the other constructors, it shouldn't matter.

Reply:
changed to init(def_conf.first, def_conf.second) and added call to execute_setup_polity(setup_policy::full)